src/test/suite/validation.test.ts

```diff
@@ -0,0 +1,107 @@
+import * as assert from 'assert';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { validate } from '../../server/messages';
+import { ServerResult } from '../../client/server_result';
+import { SOAPClient } from '../../client/SOAP_client';
+
+
+suite('Validation Test Suite', () => {
+    vscode.window.showInformationMessage('Start validation tests.');
+
+    test('Server results test for error_belsunu.atf', async() => {
+        //TODO Do we need the href html labels to make these errors clickable?
+        const expected_user_log = fs.readFileSync(path.join(
+          __dirname,'../../../src/test/suite/reference/user_log_error.log'),
+          'utf-8').trim(); // trim to avoid editor adding \n
+        const expected_val_errors = {
+            '0': 'ATF processor ox issued 2 warnings and 0 notices',
+            '6': 'unknown block token: tableta',
+            '44': 'o 4: translation uses undefined label'
+        };
+
+        // TODO Replace this with the actual content of the oracc log that
+        // comes from the server - check in Nammu
+        const oracc_log = fs.readFileSync(
+          path.join(__dirname,
+                    '../../../src/test/suite/input/error_oracc.log'),
+          'utf-8');
+        const server_result = new ServerResult(oracc_log);
+
+        // NB: The server response will always have \n as line separators!
+        assert.equal(server_result.user_log, expected_user_log);
+        // There's no sensible way to compare dictionaries, JSON.stringify
+        // seemed the most straight forward
+        assert(JSON.stringify(server_result.validation_errors) === JSON.stringify(expected_val_errors));
```

Comment:
```suggestion
        assert.strictEqual(JSON.stringify(server_result.validation_errors),  JSON.stringify(expected_val_errors));
```
IIRC, this displayed the two sides more nicely if the assertion fails? (and similar in the tests below)

Reply:
Displaying assert problems nicely is a huge benefit because at the moment it's a pain to know what's going on when assertions fail.